### ISSUE TITLE
Data Provenance: validation error lacks information

### ISSUE THREAD

**Issue by mr-c** [IQSS/dataverse#8613]
**What steps does it take to reproduce the issue?**

Attempt to add the JSON provenance from a CWLProv Research Object as data provenance

example provenance JSON file from a CWLProv Research Object: https://github.com/common-workflow-language/cwlprov/blob/main/examples/revsort-run-1/metadata/provenance/primary.cwlprov.json

* When does this issue occur?

* Which page(s) does it occurs on?
https://dataverse.nl/editdatafiles.xhtml?selectedFileIds=266295&datasetId=266294&version=6

* What happens?
Validation error without details

* To whom does it occur (all users, curators, superusers)?
At least to me

* What did you expect to happen?
Details about how the file failed validation. Links to third-party tools to help fix this.

**Which version of Dataverse are you using?**

5.6

**Any related open or closed issues to this bug report?**

**Screenshots:**
![image](https://user-images.githubusercontent.com/1330696/163409470-0bd6197c-16f5-4734-81ec-f1df045b2a93.png)

**Comment by pdurbin** [comment]
@mr-c you're right, the validation error does lack information. From a look at the code, that trailing hyphen is unintentional. It implies that there is more detail to see but the code hasn't been written to show any more information.

The popup is trying to communicate that the only provenance files that are accepted are JSON files that follow (validate, using a JSON Schema) a certain W3C standard, and links to https://www.w3.org/Submission/2013/SUBM-prov-json-20130424/ 

That CWLProv format you provided is not supported. (I tried your JSON file myself on my dev laptop.)

If you (or anyone) is interested in the details of which JSON Schema file is used for validation, please see:

- testProvValidator: https://github.com/IQSS/dataverse/blob/v5.10.1/src/test/java/edu/harvard/iq/dataverse/provenance/ProvInvestigatorTest.java#L38
- isProvValid/schema.validate: https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/provenance/ProvInvestigator.java#L117

So, my thoughts on how to fix this are:

- Remove the trailing hyphen.
- Make it extra clear that we only support one very specific prov format and that we use a JSON Schema for that format to validate against. (Perhaps link to the JSON Schema file so users can download it for offline validation.)

**Comment by pdurbin** [comment]
@mr-c out of curiosity, have you tried upload the prov file via API? Do you know if you get a more detailed error message?

Here's the API:

https://guides.dataverse.org/en/5.12/api/native-api.html#create-update-provenance-json-and-provide-related-entity-name-for-an-uploaded-file

When I'm at my desk I can run this related test and see what the output is:

https://github.com/IQSS/dataverse/blob/v5.12/src/test/java/edu/harvard/iq/dataverse/api/ProvIT.java#L143-L148

**Comment by mr-c** [comment]
@pdurbin Sorry, I'm not available to test the API; you are welcome to use the file linked from my original post.

**Comment by pdurbin** [comment]
Related:

- #9342

**Comment by pdurbin** [comment]
We just merged this PR:

- #12468

@mr-c right now https://beta.dataverse.org is down but if you check back later (and wait for another deployment) you're welcome to verify the fix!
